frontend/src/component/events/EventLog/EventActions.tsx

```diff
@@ -0,0 +1,115 @@
+import { type FC, useState } from 'react';
+import {
+    IconButton,
+    ListItemText,
+    MenuItem,
+    MenuList,
+    Popover,
+    styled,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import FileDownload from '@mui/icons-material/FileDownload';
+import type { EventSchema } from '../../../openapi';
+
+const StyledActions = styled('div')(({ theme }) => ({
+    display: 'flex',
+    justifyContent: 'center',
+}));
+
+const StyledPopover = styled(Popover)(({ theme }) => ({
+    borderRadius: theme.shape.borderRadiusLarge,
+    padding: theme.spacing(1, 1.5),
+}));
+
+interface IEventActions {
+    events: EventSchema[];
+}
+
+export const EventActions: FC<IEventActions> = ({ events }) => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const open = Boolean(anchorEl);
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    const exportJson = () => {
+        const jsonString = JSON.stringify(events);
+        const blob = new Blob([jsonString], { type: 'application/json' });
+        const url = URL.createObjectURL(blob);
+
+        const currentDate = new Date().toISOString().split('T')[0];
+        const fileName = `events_${currentDate}.json`;
+
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = fileName;
+        a.click();
+
+        URL.revokeObjectURL(url);
+        setAnchorEl(null);
+    };
```

Comment:
Is it not possible to just point to the API with the current search string and use that to generate it? Not very important, but it might save us a few lines of code 🤷🏼  Might not, though

Reply:
I would rather spend user machine resources that our servers. Also the data is already there.